packages/react/src/components/form/initial/controls.tsx

```diff
@@ -0,0 +1,407 @@
+import { Children, useEffect, useMemo, useState } from "react";
+import {
+  filterFactors,
+  getHandleTypes,
+  isFactorNonOidc,
+  isFactorOidc,
+  parsePhoneNumber,
+  resolveLastHandleValue,
+} from "../../../domain/handles";
+import { useConfiguration } from "../../../hooks/use-configuration";
+import { FactorNonOIDC, Handle, HandleType } from "../../../domain/types";
+import { TextConfig, TextConfigKey } from "../../text/constants";
+import { useInternalFormContext } from "../form";
+import { useForm } from "../../../hooks/use-form";
+import { Factor } from "@slashid/slashid";
+import { Flag, findFlag } from "country-list-with-dial-code-and-flag";
+import { sprinkles } from "../../../theme/sprinkles.css";
+import { Dropdown } from "../../dropdown";
+import { GB_FLAG, Input, PhoneInput } from "../../input";
+import { ErrorMessage } from "../error-message";
+import { isValidPhoneNumber, isValidEmail } from "../validation";
+import { Button } from "../../button";
+import { FormStatus } from "../../../context/form-context";
+import { Tabs } from "../../tabs";
+
+export const FACTOR_LABEL_MAP: Record<
+  Exclude<Factor["method"], "webauthn_via_email" | "webauthn_via_sms">,
+  TextConfigKey
+> = {
+  email_link: "factor.emailLink",
+  otp_via_sms: "factor.otpViaSms",
+  otp_via_email: "factor.otpViaEmail",
+  sms_link: "factor.smsLink",
+  webauthn: "factor.webauthn",
+  oidc: "",
+};
+
+export const TAB_NAME = {
+  email: "email",
+  phone: "phone",
+};
+
+export const tabIDByHandle: Record<HandleType, string> = {
+  phone_number: TAB_NAME.phone,
+  email_address: TAB_NAME.email,
+};
+
+type ControlsProps = {
+  factors: FactorNonOIDC[];
+  handleTypes: Handle["type"][];
+  text: TextConfig;
+  handleSubmit: (factor: Factor, handle?: Handle) => void;
+};
+
+type Props = {
+  children?:
+    | React.ReactNode
+    | (({
+        factors,
+        handleTypes,
+        text,
+        handleSubmit,
+      }: ControlsProps) => React.ReactNode);
+};
+
+/**
+ * Component responsible for rendering the form and the related controls.
+ * Sets up the form context and provides the submit handler.
+ */
+export const Controls = ({ children }: Props) => {
```

Comment:
This module has some duplication - I can do another pass refactoring this out. If you guys have any good ideas on how to structure this differently please let me know.

Reply:
I'd think of nesting this in `/controls/<component>.tsx` and exporting Controls from `index`., as well as
```
Controls.displayName = "Controls";
Controls.Input = FormInput;
Controls.Submit = Submit;
```